app/src/main/java/com/limelight/binding/input/trackpad/Core/TrackpadFramework.java

```diff
@@ -0,0 +1,786 @@
+
+    private void recodeFirstHistorical(MotionEvent motionEvent){
+        float[] rawX = axisManager(motionEvent)[0];
+        float[] rawY = axisManager(motionEvent)[1];
+
+        for(int i=0;i<motionEvent.getPointerCount();i++){
+            firstX[i] = rawX[i];
+            firstY[i] = rawY[i];
+            lastX[i] = historicalAxisX[i];
+            lastY[i] = historicalAxisY[i];
+            historicalAxisX[i] = rawX[i];
+            historicalAxisY[i] = rawY[i];
+            axis_Relative_X[i] = 0;
+            axis_Relative_Y[i] = 0;
+        }
+    }
+
+    // Relative
+    private void getRelative(MotionEvent motionEvent){
+        float[] rawX = axisManager(motionEvent)[0];
+        float[] rawY = axisManager(motionEvent)[1];
+
+        for(int i = 0; i< motionEvent.getPointerCount(); i++){
+            axis_Relative_X[i] = (rawX[i]-historicalAxisX[i]);
+            historicalAxisX[i] = rawX[i];
+            axis_Relative_Y[i] = (historicalAxisY[i]-rawY[i]);
+            historicalAxisY[i] = rawY[i];
+        }
+    }
+
+    // onePointer
+    private boolean isDrag = false;
+    private boolean single_Queue = false;
+
+    private Timer autoMoveTimer;
+    private Timer clickTimer;
+    private boolean isRight =false;
+
+    private int upDown = 0;
+    private int leftRight = 0;
+
+    private void onePointer(MotionEvent motionEvent){
+        long eventTime = SystemClock.uptimeMillis();
+        long actionUpTime = this.actionUpTime[0];
+        long actionDownTime = this.actionDownTime[0];
+        float x = axisManager(motionEvent)[0][0];
+        float y = axisManager(motionEvent)[1][0];
+        float firstX = this.firstX[0];
+        float firstY = this.firstY[0];
+        float lastX = this.lastX[0];
+        float lastY = this.lastY[0];
+        float RelativeX = axis_Relative_X[0];
+        float RelativeY = axis_Relative_Y[0];
+
+        switch (motionEvent.getActionMasked()){
+            // Touch Action
+            // Move -> Click -> Drag -> Double Click
+            case MotionEvent.ACTION_DOWN:
+                // When there is an action UP event before
+                if (single_Queue && !distanceExceeds((int)(lastX-x), (int)(lastY-y),150)) {
+                    if ((eventTime - actionUpTime) < clickToDragThreshold) { // Click -> Drag
+                        isDrag = true;
+                        if (clickTimer != null) { // TapClick Timer cancel
+                            clickTimer.cancel();
+                            clickTimer = null;
+                        }
+                    }
+                }
+                break;
+
+            // Button Action
+            case MotionEvent.ACTION_BUTTON_PRESS:
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+                    switch (motionEvent.getActionButton()){
+                        case MotionEvent.BUTTON_PRIMARY:
+                            buttonTimer = new Timer(true);
```

Comment:
Please use `Handler.postDelayed()` instead of a `Timer` to minimize creation of background threads.